go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -282,11 +282,16 @@ func runOvnKube(ctx *cli.Context) error {
 		metrics.StartMetricsServer(config.Metrics.BindAddress, config.Metrics.EnablePprof)
 	}
 
-	// start the prometheus server to serve OVN Metrics (default port: 9476)
-	// Note: for ovnkube node mode dpu-host no ovn metrics is required as ovn is not running on the node.
-	if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Metrics.OVNMetricsBindAddress != "" {
-		metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node)
-		metrics.StartOVNMetricsServer(config.Metrics.OVNMetricsBindAddress)
+	// start the prometheus server to serve OVS and OVN Metrics (default port: 9476)
+	// Note: for ovnkube node mode dpu-host no metrics is required as ovs/ovn is not running on the node.
+	if config.OvnKubeNode.Mode != types.NodeModeDPUHost {
+		if config.Metrics.OVNMetricsBindAddress != "" {
+			metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node)
+			metrics.StartOVNMetricsServer(config.Metrics.OVNMetricsBindAddress)
+		}
+		if config.Metrics.ExportOVSMetrics {
+			metrics.RegisterOvsMetricsWithOvnMetrics()
```

Comment:
We need to start OVN metrics server here by calling func `StartOVNMetricsServer` - rework this.